.github/workflows/sui-protocol-version-monitor.yml

```diff
@@ -27,37 +27,50 @@ jobs:
+        github_token: ${{secrets.GITHUB_TOKEN}}
+        workflow: sui-protocol-version-monitor.yml
+        workflow_conclusion: success
         name: ${{ inputs.environment }}-protocol-version
-    - shell: bash
+        check_artifacts: true
+        path: /tmp/${{ inputs.environment }}_saved_protocol_version.txt
+        if_no_artifact_found: warn
+
+    - name: Read last saved protocol version for ${{ inputs.environment }}
+      continue-on-error: true
+      shell: bash
       run: |
-        echo "saved_protocol_version=`cat /tmp/${{ inputs.environment }}_saved_protocol_version.txt`" >> $GITHUB_ENV
+        echo "saved_protocol_version=`cat /tmp/${{ inputs.environment }}_saved_protocol_version.txt/${{ inputs.environment }}_saved_protocol_version.txt || ''`" >> $GITHUB_ENV
 
-    - name: Read live protocol in ${{ inputs.environment }}
-      if: ${{ env.saved_protocol_version != '' }}
-      working-directory: ./
+    - name: Read live protocol version in ${{ inputs.environment }}
+      shell: bash
       run: | 
-        curl -s https://fullnode.${{ inputs.environment }}.sui.io:443 -H 'Content-Type: application/json' -d '{"jsonrpc":"2.0", "method":"suix_getLatestSuiSystemState", "params":[], "id":1}' | jq -r '.result.protocolVersion' > /tmp/${{ inputs.environment }}_current_protocol_version.txt
-        echo "current_protocol_version=`cat /tmp/${{ inputs.environment }}_current_protocol_version.txt`" >> $GITHUB_ENV
-
-    - name: Checkout sui source
-      if: ${{ env.current_protocol_version > env.saved_protocol_version }}
-      uses: actions/checkout@7dd9e2a3dc350cf687eb1b2a4fadfee8c8e49675 # pin@v3
-
+        curl -s https://fullnode.${{ inputs.environment }}.sui.io:443 -H 'Content-Type: application/json' -d '{"jsonrpc":"2.0", "method":"suix_getLatestSuiSystemState", "params":[], "id":1}' | jq -r '.result.protocolVersion' > /tmp/${{ inputs.environment }}_protocol.txt
+        echo "current_protocol_version=`cat /tmp/${{ inputs.environment }}_protocol.txt`" >> $GITHUB_ENV
 
     - name: Update framework/${{ inputs.environment }}
-      if: ${{ env.current_protocol_version > env.saved_protocol_version && github.ref ==  'refs/heads/master' }}
-      uses: Embraser01/update-git-branch-action@v1.0.0
+      if: ${{ env.saved_protocol_version != '' && env.current_protocol_version > env.saved_protocol_version }}
+      continue-on-error: true
+      uses: LucasRuy/sync-branch-action@v1.1.1
       with:
-        branch: framework//${{ inputs.environment }}
-        force: 1 # To push-force to the branch
-        githubToken: ${{ secrets.GITHUB_TOKEN }} # Github Token
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        SOURCE_BRANCH: ${{ inputs.environment }}
+        DESTINATION_BRANCH: framework/${{ inputs.environment }}
+
+    - name: Save current protocol version for ${{ inputs.environment }} to artifact
+      shell: bash
+      run: |
+        echo ${{ env.current_protocol_version }} > /tmp/${{ inputs.environment }}_protocol.txt
```

Comment:
Can we rely on this file existing from the previous action?

Reply:
This is just a temp file to store before I re-name it to the file we expect to extract in the download artifact section
```
mv /tmp/${{ inputs.environment }}_protocol.txt /tmp/${{ inputs.environment }}_saved_protocol_version.txt
```

Reply:
What I meant here was that in the previous workflow run, we create a file with the same name, and the question was whether we could re-use that file rather than create it again (I wasn't sure whether a `/tmp` file from a previous step would be accessible, but it seems like we do rely on that in some other places, so it's a potential clean-up.

Reply:
It run gets it own runner, so the `/tmp` location will not be available on the next workflow run